libsweep/src/win/serial.cc

```diff
@@ -0,0 +1,351 @@
+#include "serial.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <cstdio>
+#include <string>
+
+#include <windows.h>
+
+namespace sweep {
+namespace serial {
+
+typedef struct error {
+  const char* what; // always literal, do not free
+} error;
+
+typedef struct device {
+  HANDLE h_comm;
+  OVERLAPPED os_reader;
+  BOOL waiting_on_read;      // Used to prevent creation of new read operation if one is outstanding
+  DWORD read_timeout_millis; // timeout interval for entire read operation
+} device;
+
+// Constructor hidden from users
+static error_s error_construct(const char* what) {
+  SWEEP_ASSERT(what);
+
+  auto out = new error{what};
+  return out;
+}
+
+const char* error_message(error_s error) {
+  SWEEP_ASSERT(error);
+
+  return error->what;
+}
+
+void error_destruct(error_s error) {
+  SWEEP_ASSERT(error);
+
+  delete error;
+}
+
+static int32_t detail_get_port_number(const char* port, error_s* error) {
+  SWEEP_ASSERT(port);
+  SWEEP_ASSERT(error);
+
+  if (strlen(port) <= 3) {
```

Comment:
@daniel-j-h: In case you aren't aware: The second parameter of `strtoll` is an **output**parameter of type `char**`. indicating the first character in the string that is not part of the parsed number. It is not a way to indicate the end in a string and if you don't care how many chars are parsed, you can just pass a `nullptr` (I think I mentioned this in the original windows PR)